chainercv/evaluations/eval_detection_voc.py

```diff
@@ -0,0 +1,245 @@
+    empty_bbox = np.zeros((0, 4), dtype=np.float32)
+    empty_label = np.zeros((0,), dtype=np.bool)
+
+    # Organize predictions into Dict[l, List[bbox]]
+    pred_bboxes_list = {l: [empty_bbox for _ in six.moves.range(n_img)]
+                        for l in valid_label}
+    pred_scores_list = {l: [empty_label for _ in six.moves.range(n_img)]
+                        for l in valid_label}
+    for n in six.moves.range(n_img):
+        for l in valid_label:
+            bboxes_l = []
+            scores_l = []
+            for r in six.moves.range(pred_bboxes[n].shape[0]):
+                if l == pred_labels[n][r]:
+                    bboxes_l.append(pred_bboxes[n][r])
+                    scores_l.append(pred_scores[n][r])
+            if len(bboxes_l) > 0:
+                pred_bboxes_list[l][n] = np.stack(bboxes_l)
+                pred_scores_list[l][n] = np.stack(scores_l)
+
+    # Organize ground truths into Dict[l, List[bbox]]
+    gt_bboxes_list = {l: [empty_bbox for _ in six.moves.range(n_img)]
+                      for l in valid_label}
+    gt_difficults_list = {l: [empty_label for _ in six.moves.range(n_img)]
+                          for l in valid_label}
+    for n in six.moves.range(n_img):
+        for l in valid_label:
+            gt_bboxes_l = []
+            gt_difficults_l = []
+            for r in six.moves.range(gt_bboxes[n].shape[0]):
+                if l == gt_labels[n][r]:
+                    gt_bboxes_l.append(gt_bboxes[n][r])
+                    if gt_difficults is not None:
+                        gt_difficults_l.append(gt_difficults[n][r])
+                    else:
+                        gt_difficults_l.append(
+                            np.array(False, dtype=np.bool))
+            if len(gt_bboxes_l) > 0:
+                gt_bboxes_list[l][n] = np.stack(gt_bboxes_l)
+                gt_difficults_list[l][n] = np.stack(gt_difficults_l)
+
+    # Accumulate recacall, precison and ap
+    results = {}
+    for l in valid_label:
+        rec, prec = _pred_and_rec_cls(
+            pred_bboxes_list[l],
+            pred_scores_list[l],
+            gt_bboxes_list[l],
+            gt_difficults_list[l],
+            min_iou)
+        ap = _voc_ap(rec, prec, use_07_metric=use_07_metric)
+        results[l] = {}
+        results[l]['recall'] = rec
+        results[l]['precision'] = prec
+        results[l]['ap'] = ap
+    results['map'] = np.asscalar(np.mean(
+        [results[l]['ap'] for l in valid_label]))
+    return results
+
+
+def _pred_and_rec_cls(
+        bboxes, scores, gt_bboxes, gt_difficults, min_iou=0.5):
+    # Calculate detection metrics with respect to a class.
+    # This function is called only when there is at least one
+    # prediction or ground truth box which is labeled as the class.
+    # bboxes: List[numpy.ndarray]
+    # scores: List[numpy.ndarray]
+    # gt_bboxes: List[numpy.ndarray]
+    # gt_difficults: List[numpy.ndarray]
+
+    npos = 0
```

Comment:
Does `npos` stand for `the number of positives`? If so, `n_pos` fits our naming rules.